app/controllers/api/metrics_controller.rb

```diff
@@ -126,15 +128,16 @@ def find_service
   end
 
   def find_metric
-    @metric = @service.all_metrics.find(params[:id])
+    @metric = find_metric_by(params[:id])
   end
 
-  def build_metric
-    @metric = if params[:metric_id]
-                @service.metrics.find(params[:metric_id]).children
-              else
-                @service.metrics
-              end.build(params[:metric] || {})
+  def find_metric_by(id)
```

Comment:
The `by` suggests some key

```suggestion
  def find_metric(id)
```

Reply:
Belongs to https://github.com/3scale/porta/pull/1778